### ISSUE TITLE
LexicalDeclarations inside blocks do not force parent block to be wrapped in braces

### ISSUE THREAD

**Issue by julesbrx** [trullock/NUglify#159]
Minifing this code :
```js
function hey() {
  const hash = location.hash;
  if (hash && hash[0] === "v") {
    const unwatch = this.$watch("array", (val) => {
      if (val.length) {
        unwatch();
      }
    });
  }
}
```

With default settings :
```cs
static void Main(string[] args)
{
    var result = Uglify.Js(System.IO.File.ReadAllText("./app.js"));
    System.Console.WriteLine(result.Code);
}
```

Results in this code :
```js
// minified
function hey(){const n=location.hash;if(n&&n[0]==="v")const n=this.$watch("arrat",t=>{t.length&&n()})}

// formatted
function hey() {
  const n = location.hash;
  if (n && n[0] === "v")
    const n = this.$watch("arrat", (t) => { // this won't work
      t.length && n();
    });
}

// error => Uncaught SyntaxError: Unexpected identifier
```

The same with `let` :
`Uncaught SyntaxError: Lexical declaration cannot appear in a single-statement context`
Works with `var`, but it's not ES6 friendly

i think that when a if statement if followed by a single line compatible code (like a variable declaration), the brackets should be kept.

```js
// with brackets
function hey(){const n=location.hash;if(n&&n[0]==="v"){const n=this.$watch("array",t=>{t.length&&n()})}}

function hey() {
  const n = location.hash;
  if (n && n[0] === "v") {
    const n = this.$watch("array", (t) => {
      t.length && n();
    });
  }
}
```

**Comment by trullock** [comment]
Thanks for the bug report and helpful details, I'll take a look shortly

**Comment by trullock** [comment]
I'm not gonna be able to get around to this for a few hours, if you fancy fixing this yourself the issue is probably in AnalyzeNodeVisitor, theres various parts which strip braces, there will be one that's doing this too aggressively by not caring about const and let, which are relatively recent language additions so it's very likely this.

**Comment by julesbrx** [comment]
I managed to get around it by doing this :
```js
function hey() {
  const hash = location.hash;
  if (hash && hash[0] === "v") {
    let unwatch;
    unwatch = this.$watch("array", (val) => {
      if (val.length) {
        unwatch();
      }
    });
  }
}
```

I'm not in a rush anyway, i'm constrained by the version used in [BundleTransformer.NUglify](https://www.nuget.org/packages/BundleTransformer.NUglify/) (and i'm not competent enough to fix it).

**Comment by trullock** [comment]
As per the new subject

```
if (x) {
	const y = x;
}
while (y) {
	let z = y;
}
```

These both produce errors, basically any block with a single `let x =` or `const y =` in it succumb to this error

**Comment by trullock** [comment]
fixed in 1.9.7
